### DISCUSSION TITLE
Does duckdb receive security patches?

### DISCUSSION THREAD

**Discussion by zsmith27** [duckdb/duckdb#12893] [Q&A] [answered]
I work for a U.S. state government agency, and I need to have all software approved by our IT agency. I submitted a request to have DuckDB added to our approved software registry. Our IT agency has not yet denied my request, but I received an email stating: 

> "DuckDB was flagged as a high risk due to exposures, as it does not appear to receive security patches."

Does DuckDB receive security patches? If so, are they documented somewhere? I believe they are looking for documentation such as this: [Apache Arrow Reporting Security Issues](https://arrow.apache.org/security/). I was not able to find something similar on duckdb.org.

**Comment by szarnyasg** [comment, marked as answer]
Hi @zsmith27, thanks for reaching out.

DuckDB has frequent releases, which contain new patches to bugs that were reported. So far, we did not receive any vulnerability reports. If a vulnerability is discovered, we will release a patch version of DuckDB.

Note - DuckDB version 1.0.0 was released just a month ago, and new releases are [already scheduled](https://duckdb.org/docs/dev/release_calendar), with v1.1.0 planned for September.

**Reply by zsmith27** [reply]
Thank you for the rapid response @szarnyasg. I believe this will be helpful to pass on to our IT agency.

**Reply by michael-3701** [reply]
Are the patched security vulnerabilties listed separately from the bug fixes/changes in the release notes? Just looking for a better way to see what was patched to share with our Security team. More specifically, I want to show that you guys have addressed CVE-2024-22682. Thanks!

**Reply by ethan-wrong** [reply]
> Are the patched security vulnerabilties listed separately from the bug fixes/changes in the release notes? Just looking for a better way to see what was patched to share with our Security team. More specifically, I want to show that you guys have addressed [CVE-2024-22682](https://github.com/advisories/GHSA-fq57-m32w-cmv5). Thanks!

I'm also interested in seeing if CVE-2024-22682 has been addressed with the current version of Duckdb 1.0.0.

**Reply by Maxxen** [reply]
@michael-3701 @ethan-wrong The CVE you linked assumes that a malicious actor already has access the your machine and can "replace the normal duckdb program on the victim's machine with this malicious extension". This is not a realistic attack vector or part of any reasonable threat model - how would we protect against someone replacing the DuckDB binary itself with a tampered version? 

Note that even if a malicious actor were to drop-in a dangerous extension into the victims computer, DuckDB will refuse to load it as the extension is not signed. As outlined in the CVE [description](https://github.com/Tu0Laj1/database_test), you need to replace a victims main DuckDB binary or trick the victim into compiling a tampered version from source themselves. 

But if you can do that you might as well have your victim delete their home directory or spawn a bash shell or something much easier.

**Reply by ethan-wrong** [reply]
Thank you @Maxxen for the context and explanation of that CVE and it's relation to duckdb. Will pass this along to our security team.

**Reply by Mytherin** [reply]
That CVE is ridiculous. Perhaps we can object to it in some way.  The content of that CVE is literally "if you compile a fake `duckdb` binary that contains malicious code, then replace an existing `duckdb` binary on the users' machine with your fake `duckdb` binary, then when the user runs that binary they will execute malicious code."

> In the DuckDB database, there is a feature for writing custom extensions. Attackers can write a custom extension containing malicious commands based on the DuckDB extension template(https://github.com/duckdb/extension-template). While compiling this malicious extension, a duckdb binary program containing this malicious extension will also be generated. Attackers can replace the normal duckdb program on the victim's machine with this malicious extension. When the victim starts this malicious duckdb program and executes a specific SQL statement, the malicious command will be triggered.

This is possible with *every* program, and has nothing to do with DuckDB itself.

**Reply by Mytherin** [reply]
I've submitted a re-score request to NIST for the CVE. Apparently [bogus CVEs](https://news.ycombinator.com/item?id=37608110) are a known problem.

**Reply by Mytherin** [reply]
The CVE is now labeled as disputed.

**Reply by Mytherin** [reply]
The CVE is now fully rejected. It is not a security vulnerability in DuckDB.
